### PR TITLE
#751: add support for code search API

### PR DESCRIPTION
**Pull request by nicoulaj**
This PR adds support for search API (code search only).

### CODE REVIEW

File: github4s/shared/src/main/scala/github4s/domain/SearchCode.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    override def paramName: String  = "user"
+    override def paramValue: String = name
+  }
+
+  final case class Organization(name: String) extends SearchCodeParam {
+    override def paramName: String  = "org"
+    override def paramValue: String = name
+  }
+
+  final case class Repository(user: String, repository: String) extends SearchCodeParam {
```
Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/domain/SearchCode.scala:59]: :art: we usually use `owner/repo` as these variable names

**Review by BenFradet — APPROVED** [review]
This is looking good :+1:

**Comment by nicoulaj** [comment]
I had to cheat and increase `max-inlines` for derivation to work...

**Comment by juanpedromoreno** [comment]
Thanks @nicoulaj for your time, I will merge and release today 🙌

**Review by juanpedromoreno — APPROVED** [review]

**Pull request merged**